uk.co.compucorp.civicrm.tasksassignments/api/v3/Document.php

```diff
@@ -232,32 +214,43 @@ function civicrm_api3_document_create_multiple($params) {
  */
 function _civicrm_api3_document_create_spec(&$params) {
 
-  //default for source_contact_id = currently logged in user
-  $params['source_contact_id']['api.default'] = 'user_contact_id';
+  $activityFields = _civicrm_api_get_fields('Activity');
+  $params = array_merge($params, $activityFields);
 
-  $params['status_id']['api.aliases'] = array('activity_status');
+  $params['status_id']['api.aliases'] = ['activity_status'];
 
-  $params['assignee_contact_id'] = array(
+  $params['assignee_contact_id'] = [
     'name' => 'assignee_id',
     'title' => 'assigned to',
     'type' => 1,
     'FKClassName' => 'CRM_Activity_DAO_ActivityContact',
-  );
-  $params['target_contact_id'] = array(
+  ];
+
+  $params['target_contact_id'] = [
     'name' => 'target_id',
     'title' => 'Activity Target',
     'type' => 1,
     'FKClassName' => 'CRM_Activity_DAO_ActivityContact',
-  );
+  ];
 
-  $params['source_contact_id'] = array(
+  $params['source_contact_id'] = [
       'name' => 'source_contact_id',
       'title' => 'Activity Source Contact',
       'type' => 1,
       'FKClassName' => 'CRM_Activity_DAO_ActivityContact',
       'api.default' => 'user_contact_id',
-  );
+  ];
 
+  // replace custom data options with their column name
+  foreach ($params as $key => $param) {
+    if (substr($key, 0, 7) === 'custom_' && isset($param['column_name'])) {
```

Comment:
discussed this point with @tunbola in chat, an extra check if `$param['extends'] === 'Activity'` would add an extra layer of safety. It may not be required, but potentially could be a case where activity has a real field starting with "custom_" where we don't want this logic applied.